### PR TITLE
CI: Improve compatibility of dependency build scripts

### PR DESCRIPTION
**Pull request by TellowKrinkle**
### Description of Changes
- Adds support for using dependency build scripts targeting relative directories
- Disables rpath-based installs in macOS cmake
  Not sure why cmake defaults to using rpaths, but since rpath search paths have to be set on executables, this leads to an issue where the qt build fails because it tries to invoke rcc from the build directory, but rcc doesn't have the deps install directory set as a search path, so it can't find the zstd dylib.  Disabling rpaths means you can't relocate the deps directory (you couldn't anyways because there are non-cmake dependencies too and they aren't using rpaths), but makes all builds embed the full path to the dylib into exes so they always find the dylib properly.  (CI builds were previously unaffected because they were finding the homebrew-installed system zstd dylib and loading that instead.)
- Update Xcode to the latest one on the macOS-13 image

### Rationale behind Changes
Easier dependency builds for users

### Suggested Testing Steps
`.github/workflows/scripts/macos/build-dependencies.sh deps` see that it completes successfully

### CODE REVIEW

File: .github/workflows/scripts/macos/build-dependencies.sh [hunk 1]
```diff
@@ -153,30 +164,26 @@ patch -u src/tools/macdeployqt/shared/shared.cpp <<EOF
...
 echo "Installing Qt Image Formats..."
 tar xf "qtimageformats-everywhere-src-$QT.tar.xz"
 cd "qtimageformats-everywhere-src-$QT"
-mkdir build
-cd build
-"$INSTALLDIR/bin/qt-configure-module" ..
-make "-j$NPROCS"
-make install
-cd ../..
+cmake -B build "${CMAKE_COMMON[@]}"
```
Comment (stenzek) [inline .github/workflows/scripts/macos/build-dependencies.sh:183]: Is there a specific reason you removed the `qt-configure-module` script? That's the intended way of building qt submodules, if there's additional cmake args that need to be specified, they can be tacked onto the end.

Comment (TellowKrinkle) [reply .github/workflows/scripts/macos/build-dependencies.sh:183]: Does it mind if I re-specify `CMAKE_INSTALL_PREFIX`, etc?

Comment (stenzek) [reply .github/workflows/scripts/macos/build-dependencies.sh:183]: I don't see why not.

**Review by stenzek — APPROVED** [review]
Tested locally, works fine. Merging so I can rebase #10773.

**Pull request merged**